include/okapi/impl/device/distanceSensor.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "api.h"
+#include "okapi/api/control/controllerInput.hpp"
+#include "okapi/api/filter/passthroughFilter.hpp"
+#include <memory>
+
+namespace okapi {
+class DistanceSensor : public ControllerInput<double> {
+  public:
+  /**
+   * A distance sensor on a V5 port.
+   *
+   * ```cpp
+   * auto ds = DistanceSensor(1);
+   * auto filteredDistSensor = DistanceSensor(1, std::make_unique<MedianFilter<5>>());
+   * ```
+   *
+   * @param iport The V5 port the device uses.
+   * @param ifilter The filter to use for filtering the distance measurements.
+   */
+  DistanceSensor(std::uint8_t iport,
+                 std::unique_ptr<Filter> ifilter = std::make_unique<PassthroughFilter>());
+
+  virtual ~DistanceSensor();
+
+  /**
+   * @return The current filtered sensor value in mm.
```

Comment:
This needs a title to be parsed by doxygen.

Reply:
Only for the overview. Do you really care about that?

Reply:
I know you already fixed it, but yeah, I care about that. It gives information at a glance about the purpose of each function and also puts the title in a clear place in the actual documentation section.